src/resonanceReconstruction/rmatrix/src/makeReichMooreChannelData.hpp

```diff
@@ -50,58 +47,21 @@ makeReichMooreChannelData(
                true,
                [] ( bool left, bool right ) { return left and right; } );
   };
-  auto retrieveQuantumNumber =
-       [&] ( unsigned int l, double j,
-             std::vector< ChannelQuantumNumbers >& available ) {
-
-    // retrieve the available numbers with this l,J
-    auto filter =
-         [l,j] ( const auto& number )
-               { return ( number.orbitalAngularMomentum() == l ) and
-                        ( number.totalAngularMomentum() == std::abs( j ) ); };
-    auto filtered = available | ranges::view::filter( filter );
-
-    auto found = ranges::distance( filtered );
-    if ( found > 0 ) {
-
-      // retrieve the one we need and erase it from the available numbers
-      ChannelQuantumNumbers numbers( 0, 0., 0., +1 );
-      if ( found == 1 ) {
-
-        numbers = filtered.front();
-      }
-      else {
-
-        numbers =  filtered.front().spin() < filtered.back().spin() ?
-                   filtered.front() : filtered.back();
-      }
-      available.erase( std::find_if(
-                           available.begin(), available.end(),
-                           [&] ( const auto& value )
-                               { return compareNumbers( value, numbers ); } ) );
-      return numbers;
-    }
-    else {
-
-      throw std::runtime_error( "None of the expected spin groups has l="
-                                + std::to_string( l ) + " and J="
-                                + std::to_string( j ) );
-    }
-  };
 
-  // the fission and capture particle pair
-  ParticlePair pFission( pair.particle(), pair.residual(),
+  // the elastic, fission and capture particle pair
+  ParticlePair in( Particle( incident, neutronMass,
+                             0.0 * coulombs, 0.5, +1),
+                   Particle( target,
+                             awri * neutronMass,
+                             0.0 * coulombs, spin, +1) );
+  ParticlePair pFission( in.particle(), in.residual(),
                          ParticlePairID( "fission" ) );
-  ParticlePair pCapture( pair.particle(), pair.residual(),
+  ParticlePair pCapture( in.particle(), in.residual(),
                          ParticlePairID( "capture" ) );
 
   // the channel radii to be used in the channels
-  ChannelRadii useRadii = radii;
-  if ( apl != 0. ) {
-
-    useRadii = makeChannelRadii( apl, nro, naps,
-                                 awri, pair.particle().mass().value );
-  }
+  ChannelRadii radii = makeChannelRadii( apl != 0. ? apl : ap, nro, naps,
```

Comment:
Nice to see good uses of ternary conditional. That's one of my favorites.

Reply:
I've been using these more and more frequently now, I don't know why.

Reply:
Maybe I'm having a bad influence on you? :-)

Reply:
Oh, you mean he co-opted MY reason?? :-D